### PR TITLE
support darwin

### PR DESCRIPTION
**Pull request by nkbai**

### CODE REVIEW

**Comment by hidez8891** [comment]
Why did you modify the test case?
If you want to fix the test case, please submit other Pull Req.

File: shmi_darwin.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+		return -1;
+	}
+	struct stat mapstat;
+	if (-1 != fstat(fd, &mapstat) && mapstat.st_size == 0) {
+		if (ftruncate(fd, size) != 0) {
+		close(fd);
+		return -2;
+		}
+	}
+	return fd;
```
Comment (hidez8891) [inline shmi_darwin.go]: What happens if `fstat` returns `-1` ?

Comment (nkbai) [reply shmi_darwin.go]: it should return an error

**Comment by nkbai** [comment]
> Why did you modify the test case?
> If you want to fix the test case, please submit other Pull Req.

why defer in for loop？
Do you just want to keep the file opened?
I did a test on the mac, and started to report an error when the size was 8192, and returned -1. I guess it was because shm_open opened too many files

File: shmi_darwin.go (hunk 1 above)
Comment (nkbai) [reply shmi_darwin.go]: it should closed file and return error

Comment (nkbai) [reply shmi_darwin.go]: This is done because the shm_open api for mac is slightly different from linux.

**Comment by hidez8891** [comment]
>why defer in for loop？
Do you just want to keep the file opened?

It is my mistake.
I will fix it.

**Pull request merged**